packages/core/test/application_init_spec.ts

```diff
@@ -73,5 +74,55 @@ import {inject, TestBed, waitForAsync} from '../testing';
            });
          })));
     });
+
+    describe('with app initializers represented using observables', () => {
+      let resolve: Subscriber<any>;
+      let observable: Observable<any>;
+      let completerResolver = false;
+      beforeEach(() => {
+        observable = new Observable((res) => {
+          resolve = res;
+        });
+        TestBed.configureTestingModule({
+          providers: [
+            {provide: APP_INITIALIZER, multi: true, useValue: () => observable},
+          ]
+        });
+      });
+
+      it('should update the status once all async observable initializers are done',
+         waitForAsync(inject([ApplicationInitStatus], (status: ApplicationInitStatus) => {
+           (status as any).runInitializers();
```

Comment:
I'd propose to add a quick comment on why we use `as any` here (and in the second test as well):
```suggestion
           // Accessing internal `runInitializers` function of the `ApplicationInitStatus` class instance
           // for testing purposes to invoke initializer functions.
           (status as any).runInitializers();
```